joiner.go

```diff
@@ -22,7 +22,8 @@ type WebSocketJoiner struct {
 	DialTimeout time.Duration
 }
 
-func (w *WebSocketJoiner) Join(ctx context.Context, url, realm string) (BaseSession, error) {
+func (w *WebSocketJoiner) Join(ctx context.Context, url, realm string, keepaliveInterval time.Duration,
```

Comment:
those new parameters should be moved inside WSDialerConfig struct. And this function should signature should that as last argument